x/cash/dynamicfee.go

```diff
@@ -65,7 +65,8 @@ func (d DynamicFeeDecorator) Check(ctx weave.Context, store weave.KVStore, tx we
 
 	defer func() {
 		if cerr == nil {
-			cache.Write()
+			//TODO: Panic?
+			_ = cache.Write()
```

Comment:
I am not sure what can cause this error to happen but IMHO we should try to ensure KV store consistency. Especially on Deliver L101
How about returning the error instead?
```
			if err:= cache.Write(); err!=nil{
				cerr = err
				return
			}
```